metacat-main/src/main/java/com/netflix/metacat/main/services/MetadataService.java

```diff
@@ -37,15 +42,39 @@
  * Created by amajumdar on 9/26/16.
  */
 @Slf4j
-@AllArgsConstructor
 public class MetadataService {
     private final Config config;
     private final TableService tableService;
     private final PartitionService partitionService;
     private final UserMetadataService userMetadataService;
+    private final TagService tagService;
+    private final MetacatServiceHelper helper;
     private final Registry registry;
 
     /**
+     * Constructor.
+     * @param config                configuration
+     * @param tableService          table service
+     * @param partitionService      partition service
+     * @param userMetadataService   usermetadata service
+     * @param tagService            tag service
+     * @param helper                service helper
+     * @param registry              registry
+     */
+    public MetadataService(final Config config, final TableService tableService,
+        final PartitionService partitionService,
+        final UserMetadataService userMetadataService, final TagService tagService,
+        final MetacatServiceHelper helper, final Registry registry) {
+        this.config = config;
+        this.tableService = tableService;
+        this.partitionService = partitionService;
+        this.userMetadataService = userMetadataService;
+        this.tagService = tagService;
+        this.helper = helper;
+        this.registry = registry;
+    }
+
+    /**
      * Deletes all the data metadata marked for deletion.
      */
     public void processDeletedDataMetadata() {
```

Comment:
I am not sure if this function is trying to delete the data metadata or process the deleted data. The comments and the function name confused me.

Reply:
It is deleting obsolete data metadata. I agree the method name is confusing. This was written long back.